Yborgess aro eap howto

yersan opened the pull request:
DRAF

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+This guide demonstrates how to deploy a Jakarta EE 8 application running on Red Hat JBoss Enterprise Application Platform (JBoss EAP) that connects to a Microsoft SQL Server to an Azure Red Hat OpenShift (ARO) 4 cluster by using [EAP Helm Charts](https://jbossas.github.io/eap-charts).
+
+The guide takes a traditional Jakarta EE 8 application and drives you by the process of migrating it to a container orchestrator like Red Hat OpenShift running on Azure. First, it describes how you can package your application as a [Bootable Jar](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html/using_jboss_eap_xp_3.0.0/the-bootable-jar_default)  ...
```
s/drives you by/walks you through/

tomjenkinson replied:
s/[Bootable Jar]/[Bootable JAR]/

yersan replied:
fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+The application is a stateful application that stores information on the HTTP Session. It makes use of the JBoss EAP clustering capabilities and uses the following Jakarta EE 8 and Microprofile 4.0 technologies:
+* Jakarta Server Faces
+* Enterprise Java Beans
```
This is now called Jakarta Enterprise Beans: "https://jakarta.ee/specifications/enterprise-beans/"

yersan replied:
Fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+The application is a stateful application that stores information on the HTTP Session. It makes use of the JBoss EAP clustering capabilities and uses the following Jakarta EE 8 and Microprofile 4.0 technologies:
+* Jakarta Server Faces
+* Enterprise Java Beans
+* Java Persistence
```
This is now called Jakarta Persistence: https://jakarta.ee/specifications/persistence/

yersan replied:
Fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+* Jakarta Server Faces
+* Enterprise Java Beans
+* Java Persistence
+* Microprofile Health
```
MicroProfile (rather than Microprofile)

tomjenkinson replied:
FYI: This needs to be changed in a few places in the document

yersan replied:
Fixed, reviewed in all the document

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+> Azure Red Hat OpenShift requires a minimum of 40 cores to create and run an OpenShift cluster. The default Azure resource quota for a new Azure subscription does not meet this requirement. To request an increase in your resource limit, see [Standard quota: Increase limits by VM series](../azure-portal/supportability/per-vm-quota-requests.md). Note that the free trial subscription isn't eligible ...
+
+1. Prepare a local machine with Unix-like operating system installed (for example, Ubuntu, macOS).
+1. Install a Java SE implementation (for example, [AdoptOpenJDK OpenJDK 11 LTS/OpenJ9](https://adoptopenjdk.net/?variant=openjdk11&jvmVariant=openj9)).
```
AdoptOpenJDK is not supported by EAP: https://access.redhat.com/solutions/4077471

yersan replied:
Fixed, added Oracle JDK with its link

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+## Prepare the application
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from you ...
```
I am not sure what is intended by "Usually, you could have a a JBoss EAP server..."?

Maybe in context here and to satisfy some assumption it would be something like, "In order to test the application locally, it is also expected that you already have a JBoss EAP server..."

(also a bit later in the statement a mix of plural and non-plural is present "drivers" vs "data source" - I am guessing here that "driver" is more accurate in that you would only need one of these adding to the server?)

tomjenkinson replied:
Assuming it's necessary, perhaps some statement about expecting EAP XP to have been installed would be added around here?

yersan replied:
Definitely, I need to review this paragraph. The intention was to emphasize the differences you could have when you are developing a "traditional" (no OpenShift targeted application) vs an application you want to run in OpenShift. 

When you are developing an application you have your EAP server installed and configured, and also a database server available on your local LAN (or local machine). But when you are targeting OpenShift, you now need to have in OpenShift a server configured with the database driver and datasource and probably you want to trim the capabilities.

I'm going to review this and try to go straight to the point.

tomjenkinson replied:
OK, I understand better now the intention, thank you for clarifying. Maybe even changing "Usually" to "Traditionally" or something gives the impression of historical events

yersan replied:
I've rephrased a bit the whole paragraph. Is more understandable now?

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from you ...
+
+When you are targeting OpenShift, you might want to trim the capabilities of your JBoss EAP server, for example, to reduce the security exposure of the provisioned server and reduce the overall footprint. You could also want to include some Microprofile specs to make your application more suitable for running on an OpenShift environment. When using JBoss EAP, one way to accomplish this is by pack ...
```
I am not sure how including MicroProfile specs would make an application more suitable for running on an OpenShift environment?

yersan replied:
Suppose the following, you have several replicas of your application, and each replica is running on a pod on a different worker node in OpenShift. Now, suppose that one of those nodes loses the ability to get a connection from the database pod. If you have not added a health check that verifies your application instance can connect to your database server, then when a client sends a request for your application, the load balancer would use any instance. If the instance that receives the client request was the one that is out of database connection, you will get an error.

However, having a health check your application is telling other parties that this specific instance is not ready to receive requests, so OpenShift will deliver the client request to a ready instance and your clients will not get an error.

That was a complex example, but, just to make an idea. Simpler examples are: you want your application to respond when it is running above some CPU or memory threshold. Heath checks can do that. If the current pod is running out of resources, then, maybe there are other pods that can satisfy the client's request.

In a container orchestrator, it is important to give your application the ability to inform third parties when the application is ready and live, and that's is available by using MicroProfile Specs. So adding MP specs makes your application more suitable to run in OpenShift.

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+1. We have added the `wildfly-jar-maven` plugin to provision the server and the application in a single executable JAR file. The OpenShift deployment unit will be now our server with our application.
+1. On the maven plugin, we have specified a set of Galleon layers. This configuration allows us to trim the server capabilities to only what we need.
+1. Our application uses JSF with Ajax requests, which means there will be information stored in the HTTP Session. We don't want to lose such information if a pod is removed. We could save this information on the client and send it back on each request. However, there are cases where you may not distribute certain information to the clients. For this demo, we have chosen to replicate the session a ...
```
s/JSF/Jakarta Faces/

tomjenkinson replied:
s/However, there are cases where you may not distribute certain information to the clients./However, there are cases where you may decide not to distribute certain information to the clients./

yersan replied:
Fixed, both

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+1. Install [Maven](https://maven.apache.org/download.cgi) 3.6.3 or higher.
+1. Install [Docker](https://docs.docker.com/get-docker/) for your OS.
+1. Install [Azure CLI](/cli/azure/install-azure-cli) 2.29.2 or later.
+1. Clone the code for this demo application on your local system. The demo application is on [GitHub](https://github.com/yersan/todo-list).
```
I think `todo-list` should appear in the text. Maybe change "Clone the code for this demo application on your local system." to "Clone the code for this demo application (todo-list) on your local system."

Also rather than "on your local system" maybe "to your local system"

yersan replied:
Right!, fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+## Deploy to OpenShift
+
+Before deploying the demo application on OpenShift we will deploy the database server. The database server will be deployed by using a [DeploymentConfig OpenShift API resource](https://docs.openshift.com/container-platform/4.8/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are). The database server deployment configuration is available as a Y ...
```
This is a guess - s/YALM/YAML/ - am I correct?

yersan replied:
Absolutely correct, fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+Before deploying the demo application on OpenShift we will deploy the database server. The database server will be deployed by using a [DeploymentConfig OpenShift API resource](https://docs.openshift.com/container-platform/4.8/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are). The database server deployment configuration is available as a Y ...
+
+> [!NOTE]
+> You can also use the JBoss EAP Operator to deploy this example, however, notice that the JBoss EAP Operator will deploy the application as stateful sets. Use the JBoss EAP Operator if your application requires one or more one of the following.
```
It could be implied within the below list (stable, persistent storage) but Transactions are a notable reason to use the Operator, I think it should be added

bstansberry replied:
+1

People don't understand whether they need stable, persistent storage and likely think they don't.

yersan replied:
I've added an entry about the transaction facility.

tomjenkinson replied:
I don't have the ability to mark this as resolved but this area could be marked as resolved for me

yersan replied:
Personally, I would prefer to have the conversations open and make the last comment count. If you have anything else to add for applying a correction, just add it to the conversation, and I'll replay when done. This is just a personal suggestion to work with a review.

tomjenkinson replied:
I am happy with however you prefer to work with this review. What I suppose I am saying is that you have addressed my initial feedback as Transaction recovery is now listed below

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+    ![OpenShift console EAP Helm Charts](./media/howto-deploy-java-eap-app/console_eap_helm_charts.png)
+
+5. Since our application uses Microprofile capabilities, you have to select for this demo the one named EAP Xp (at the time of this writing, the exact version is EAP Xp 1.0.0). The `Xp` stands for Expansion Pack. With the JBoss Enterprise Application Platform expansion pack, developers can use Eclipse MicroProfile application programming interfaces (APIs) to build and deploy microservices-based a ...
```
The XP casing is incorrect here: `s/EAP Xp/EAP XP/g` and the current version is EAP XP 3.0 (https://access.redhat.com/articles/2026253#CompatibleEAPXPVersions74)

tomjenkinson replied:
Incorrect `Xp` casing exists elsewhere in the document

tomjenkinson replied:
I think there is some incorrect casing on the OpenShift Helm Chart from your screen shot and this should be reported elsewhere and updated. But there are changes that should be made in this document too. According to your screen shot the version is "Eap Xp3 v1.0.0" (please note that there is the presence of 3 in that version)

bstansberry replied:
I have a vague recollection that the casing used in the console UI is out of our control. @yersan , if you don't already know if I'm right about that, Jeff would know.

yersan replied:
Jeff confirmed it is out of our control, there is the same issue with WildFly, which is rendered as Wildfly. OpenShift decides how it gets rendered.

In any case, I've added the missing '3'

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+1. Clone the code for this demo application on your local system. The demo application is on [GitHub](https://github.com/yersan/todo-list).
+1. Follow the instructions in [Create an Azure Red Hat OpenShift 4 cluster](./tutorial-create-cluster.md).
+
+   Though the "Get a Red Hat pull secret" step is labeled as optional, **it is required for this article**.  The pull secret enables your Azure Red Hat OpenShift cluster to find the JBoss EAP application images.
```
I am thinking that the use of "Azure Red Hat OpenShift" could be changed to the acronym ARO for this specific instance as it is normal text and ARO has been defined.

yersan replied:
Sounds good, changed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+    The Helm Release (abbreviated HR) is named `eap-todo-list-demo`. It includes a Deployment resource (abbreviated D) also named `eap-todo-list-demo`. 
+
+1. When build is finished (the bottom-left icon will display a green check) and the application is deployed (the circle outline is in dark blue), you can go to application the URL (using the top-right icon) from the route associated to the deployment.
```
s/When build/When the build/

yersan replied:
Fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+    ![OpenShift application running](./media/howto-deploy-java-eap-app/application_running_openshift.png)
+
+1. The application shows you the name of the pod which has served the information. To verify the clustering capabilities you could add some Todos, change the filter to see all the records and mark some of them as completed. Then delete the pod name that appears on the application, and once deleted, create a new Todo on the same application window. You will see that the new Todo is added via an aj ...
```
Previously Ajax was capitalized but in this paragraph it is not. I am not sure which way is correct.

tomjenkinson replied:
What is intended by "delete the pod name" - is this rather to `delete the pod with the name indicated in the "Server Host Name" field' - perhaps you would give some information on how to delete the pod or a reference?

tomjenkinson replied:
If I followed correctly: `s/same list/expected list, including the new Todo/`

yersan replied:
> Previously Ajax was capitalized but in this paragraph it is not. I am not sure which way is correct.

Ajax or AJAX are both fine, ajax is not correct. Fixed.

In addition, this discussion raised some questions about how clustering is working for JSF. I need to review this area again, I'll let you know when done.

yersan replied:
Done, I've revisited this paragraph describing better what is happening with the cluster.

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+Locate the `eap-todo-list-demo` Helm Chart and at the end of the row, click on the tree dots to open the action contextual menu entry.
+
+Click on "Uninstal Helm Release" to remove the application. Notice the application config map is not part of the chart and you need to remove it separately if you no longer need it.
```
s/Uninstal/Uninstall/

yersan replied:
Fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+Click on "Uninstal Helm Release" to remove the application. Notice the application config map is not part of the chart and you need to remove it separately if you no longer need it.
+
+Execute the following command if you want to delete the configmap that hold the application configuration:
```
s/that hold/that holds/

yersan replied:
Fixed, also configmaps were replaced by secrets

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+## Next steps
+
+In this guide, you learned how to:
+> [!div class="checklist"]
```
This `[!div class="checklist"]` is being rendered - is it expected or should the line and the following line be removed?

yersan replied:
sorry, I should have mentioned before, to correctly display the final document, you need to do it by using Visual Studio Code with the Azure documentation tool package installed: https://marketplace.visualstudio.com/items?itemName=docsmsft.docs-authoring-pack

[!div class="checklist"] renders fine in Visual Studio with the doc pack installed.

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+This guide demonstrates how to deploy a Jakarta EE 8 application running on Red Hat JBoss Enterprise Application Platform (JBoss EAP) that connects to a Microsoft SQL Server to an Azure Red Hat OpenShift (ARO) 4 cluster by using [EAP Helm Charts](https://jbossas.github.io/eap-charts).
+
+The guide takes a traditional Jakarta EE 8 application and drives you by the process of migrating it to a container orchestrator like Red Hat OpenShift running on Azure. First, it describes how you can package your application as a [Bootable Jar](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html/using_jboss_eap_xp_3.0.0/the-bootable-jar_default)  ...
```
s/locally connecting/locally, connecting/g

yersan replied:
fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+The guide takes a traditional Jakarta EE 8 application and drives you by the process of migrating it to a container orchestrator like Red Hat OpenShift running on Azure. First, it describes how you can package your application as a [Bootable Jar](https://access.redhat.com/documentation/en-us/red_hat_jboss_enterprise_application_platform/7.4/html/using_jboss_eap_xp_3.0.0/the-bootable-jar_default)  ...
+
+The application is a stateful application that stores information on the HTTP Session. It makes use of the JBoss EAP clustering capabilities and uses the following Jakarta EE 8 and Microprofile 4.0 technologies:
```
s/information on the/information in a/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+## Prerequisites
+
+> [!NOTE]
+> Azure Red Hat OpenShift requires a minimum of 40 cores to create and run an OpenShift cluster. The default Azure resource quota for a new Azure subscription does not meet this requirement. To request an increase in your resource limit, see [Standard quota: Increase limits by VM series](../azure-portal/supportability/per-vm-quota-requests.md). Note that the free trial subscription isn't eligible ...
```
Either

s/increase, [upgrade/increase; [upgrade/g

or 

s/increase, [upgrade/increase. [Upgrade/g

Probably the latter is better.

This sounds like standard language you got from somewhere else though, so if that's true and you're supposed to just use that language unchanged, that's fine.

yersan replied:
Yes, part of the pre-requisite steps were taken from an existing guide available at the Azure docs. Per your feedback, I leave it as it is now.

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+> [!NOTE]
+> Azure Red Hat OpenShift requires a minimum of 40 cores to create and run an OpenShift cluster. The default Azure resource quota for a new Azure subscription does not meet this requirement. To request an increase in your resource limit, see [Standard quota: Increase limits by VM series](../azure-portal/supportability/per-vm-quota-requests.md). Note that the free trial subscription isn't eligible ...
+
+1. Prepare a local machine with Unix-like operating system installed (for example, Ubuntu, macOS).
```
s/with Unix/with a Unix/g

Shouldn't RHEL and maybe Fedora be in the list?

yersan replied:
Why not?, added RHEL and Fedora and removed Ubuntu

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+1. Install [Maven](https://maven.apache.org/download.cgi) 3.6.3 or higher.
+1. Install [Docker](https://docs.docker.com/get-docker/) for your OS.
+1. Install [Azure CLI](/cli/azure/install-azure-cli) 2.29.2 or later.
+1. Clone the code for this demo application on your local system. The demo application is on [GitHub](https://github.com/yersan/todo-list).
```
s/on your/to your/g

bstansberry replied:
s/application is on/application is at/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+## Prepare the application
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from you ...
```
s/on your environment/in your environment/g

yersan replied:
No longer applicable.

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+At this stage, you have cloned the demo application and your local repository is on the `main` branch. The demo application is a simple Jakarta EE 8 application that creates, reads, updates, and deletes records on a Microsoft SQL Server. Usually, you could have a JBoss EAP server installed on your local machine and configured with the required drivers and data source to allow connections from you ...
+
+When you are targeting OpenShift, you might want to trim the capabilities of your JBoss EAP server, for example, to reduce the security exposure of the provisioned server and reduce the overall footprint. You could also want to include some Microprofile specs to make your application more suitable for running on an OpenShift environment. When using JBoss EAP, one way to accomplish this is by pack ...
```
s/server, for example,/server; for example,/g

s/as Bootable/as a Bootable/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+1. We have added the `wildfly-jar-maven` plugin to provision the server and the application in a single executable JAR file. The OpenShift deployment unit will be now our server with our application.
+1. On the maven plugin, we have specified a set of Galleon layers. This configuration allows us to trim the server capabilities to only what we need.
+1. Our application uses JSF with Ajax requests, which means there will be information stored in the HTTP Session. We don't want to lose such information if a pod is removed. We could save this information on the client and send it back on each request. However, there are cases where you may not distribute certain information to the clients. For this demo, we have chosen to replicate the session a ...
```
s/on the `web.xml`/to the `web.xml`/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+    MSSQLSERVER_DATABASE=todos_db \
+    MSSQLSERVER_HOST=localhost \
+    MSSQLSERVER_PORT=1433 \
+    java -jar ./target/todo-list-bootable.jar -Djboss.node.name=node2 -Djboss.socket.binding.port-offset=10
```
You use an offset of 10 here but in the example below you use 9080 for the http port, so an offset of 1000.

yersan replied:
Good catch, thanks!, fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+    ![ToDo EAP demo Application](./media/howto-deploy-java-eap-app/todo-demo-application.png)
+
+1. Check the application health endpoints. These endpoints will be used by OpenShift to verify when your pod is live and ready to receive user requests:
```
You say 'live and ready' but the example is a liveness check, not a readiness check.

yersan replied:
I added both at the curl commands. I've added a blank line in between the health checks curl commands to avoid confusion.

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+## Deploy to OpenShift
+
+Before deploying the demo application on OpenShift we will deploy the database server. The database server will be deployed by using a [DeploymentConfig OpenShift API resource](https://docs.openshift.com/container-platform/4.8/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are). The database server deployment configuration is available as a Y ...
```
s/file on the application source/file in the application source/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+## Deploy to OpenShift
+
+Before deploying the demo application on OpenShift we will deploy the database server. The database server will be deployed by using a [DeploymentConfig OpenShift API resource](https://docs.openshift.com/container-platform/4.8/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are). The database server deployment configuration is available as a Y ...
```
I think the part beginning with "To deploy the application" should be a new paragraph.

yersan replied:
I agree, fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+## Deploy to OpenShift
+
+Before deploying the demo application on OpenShift we will deploy the database server. The database server will be deployed by using a [DeploymentConfig OpenShift API resource](https://docs.openshift.com/container-platform/4.8/applications/deployments/what-deployments-are.html#deployments-and-deploymentconfigs_what-deployments-are). The database server deployment configuration is available as a Y ...
```
s/sensible/sensitive

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+Let's do a quick review about what we have changed:
+
+1. We have added a new maven profile name `bootable-jar-openshift` that prepares the Bootable JAR with a specific configuration for running the server on the cloud, for example, it enables the JGroup subsystem to use TCP requests to discover other pods by using the KUBE_PING protocol.
```
s/name/named

s/cloud, for example/cloud; for example/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+### Deploy the application to OpenShift
+
+Now that we have the database server ready, we can deploy the demo application via JBoss EAP Helm Charts. The Helm Chart application configuration file is available at `todo-list/deployment/application/todo-list-helm-chart.yaml`. You could deploy this file via command line, however, to do so you would need to have Helm Charts installed on your local machine. Instead of using the command line, the ...
```
s/via command line/via the command line/g

s/line, however/line; however/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+Now that we have the database server ready, we can deploy the demo application via JBoss EAP Helm Charts. The Helm Chart application configuration file is available at `todo-list/deployment/application/todo-list-helm-chart.yaml`. You could deploy this file via command line, however, to do so you would need to have Helm Charts installed on your local machine. Instead of using the command line, the ...
+
+Before deploying the application, let's creating the expected secret that will hold specific application configuration. The Helm Chart will get the database user, password and name from the `mssqlserver-secret`secret object created before, and the driver version, the datasource JNDI name and the cluster password from the following secret:
```
s/creating/create

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+    > You decide which cluster password you want to use, the pods that want to join to your cluster need such password, that prevents your JBoss EAP cluster is not messed up with any pod that are not under your control and ask to join.
+
+    > [!NOTE]
+    > You could have noticed from the above config map that we are not supplying the database Hostname and Port. That's not necessary. If you take a closer look at the application Helm Chart file, you will see that the database Hostname and Port are passed by using the following notations \$(MSSQLSERVER_SERVICE_HOST) and \$(MSSQLSERVER_SERVICE_PORT). This is a standard OpenShift notation that wil ...
```
s/could have noticed/may have noticed/g

s/notation that will make/notation that will ensure/g

s/values of pod/values of the pod/g

s/available/that are available/g

s/, which are passed/. Pod environment variable are passed/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+At this point, we need to configure the chart to be able to build and deploy the application:
+
+1. Change the name of the release to `eap-todo-list-demo`. 
+1. We can configure the Helm Chart either using a `Form View` or a `YAML View`. Check `YAML View` radio button in the `Configure via` box.
```
s/Check/Check the/g

yersan replied:
Fixed

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+    ![OpenShift application running](./media/howto-deploy-java-eap-app/application_running_openshift.png)
+
+1. The application shows you the name of the pod which has served the information. To verify the clustering capabilities you could add some Todos, change the filter to see all the records and mark some of them as completed. Then delete the pod name that appears on the application, and once deleted, create a new Todo on the same application window. You will see that the new Todo is added via an aj ...
```
s/create a new Todo on/create a new Todo in/g

s/different pod, indeed/different pod. Indeed/g

yersan replied:
Fixed, although I need to revisit the paragraph.

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,384 @@
+
+### Delete the database
+
+If you only want to delete the database and the related objects, execute the following command:
```
Nit: "only" in this sentence implies you'd perhaps drop the db but leave the app, which sounds odd.

yersan replied:
Yes, right, although if you are testing things, maybe you want to know how to delete the database resources. In any case, yes, sounds a bit odd, so I removed the "only"

yersan commented:
Hi @tomjenkinson  / @bstansberry I've addressed most of the issues. When I was reviewing the comments, I arrived at some details about clustering that I still don't understand at all, and I need to be sure I understand it correctly. I would like to play a bit more with it and let you know when done for a second review. Thank you!

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -198,6 +208,7 @@ Before deploying the demo application on OpenShift we will deploy the database s
 > - Stable, persistent storage.
 > - Ordered, graceful deployment and scaling.
 > - Ordered, automated rolling updates.
+> - Transaction recovery facility when a pod is scaled down.
```
s/scaled down/scaled down or crashes/

yersan replied:
Done

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -31,15 +31,15 @@ The application is a stateful application that stores information on the HTTP Se
 
-1. Prepare a local machine with Unix-like operating system installed (for example, Ubuntu, macOS).
-1. Install a Java SE implementation (for example, [AdoptOpenJDK OpenJDK 11 LTS/OpenJ9](https://adoptopenjdk.net/?variant=openjdk11&jvmVariant=openj9)).
+1. Prepare a local machine with Unix-like operating system installed (for example, Fedora, RHEL    , macOS).
```
I think there could be some extra spaces after RHEL that are not intended?

yersan replied:
Ouhh, yes, my fault, fixed

tomjenkinson commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -277,31 +288,31 @@ Before deploying the application, let's creating the expected secret that will h
+1. We can configure the Helm Chart either using a `Form View` or a `YAML View`. Check the `YAML View` radio button in the `Configure via` box.
 1. Then, change the YAML content to configure the Helm Chart by copying the content of the Helm Chart file available at `todo-list/deployment/application/todo-list-helm-chart.yaml` instead of the existing content:
 
    ![OpenShift console EAP Helm Chart YAML content](./media/howto-deploy-java-eap-app/console_eap_helm_charts_yalm_content.png)
```
I can see here "console_eap_helm_charts_yalm_content" - I think this should have yaml instead of yalm and I think https://github.com/yersan/azure-docs/blob/yborgess-aro-eap-howto/articles/openshift/media/howto-deploy-java-eap-app/console_eap_helm_charts_yalm_content.png needs to be renamed accordingly?

yersan replied:
Great catch! thanks, renamed.

tomjenkinson reviewed (approved):
Great blog - thank you!

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,395 @@
+
+# Deploy a Java application with Red Hat JBoss Enterprise Application Platform on an Azure Red Hat OpenShift 4 cluster
+
+This guide demonstrates how to deploy a Jakarta EE 8 application running on Red Hat JBoss Enterprise Application Platform (JBoss EAP) that connects to a Microsoft SQL Server to an Azure Red Hat OpenShift (ARO) 4 cluster by using [EAP Helm Charts](https://jbossas.github.io/eap-charts).
```
I struggled with this sentence yesterday but didn't have any better ideas that didn't amount kind of duplicating the next paragraph. And I think a single sentence summary is a good thing.

Maybe this will help...

Put a comma after "Jakarta EE 8 application" and after "Microsoft SQL Server".  That separates off the entire "running on Red Hat JBoss Enterprise Application Platform (JBoss EAP) that connects to a Microsoft SQL Server" part, which helps. That block can be removed entirely and you have a grammatically correct and accurate sentence. But of course you don't want to remove it! So separating it a bit with commas might help.

yersan replied:
Done

bstansberry commented on articles/openshift/howto-deploy-java-eap-app.md:
```diff
@@ -0,0 +1,395 @@
+> [!NOTE]
+> Azure Red Hat OpenShift requires a minimum of 40 cores to create and run an OpenShift cluster. The default Azure resource quota for a new Azure subscription does not meet this requirement. To request an increase in your resource limit, see [Standard quota: Increase limits by VM series](../azure-portal/supportability/per-vm-quota-requests.md). Note that the free trial subscription isn't eligible ...
+
+1. Prepare a local machine with Unix-like operating system installed (for example, Fedora, RHEL, macOS).
```
s/with Unix/with a Unix/g

yersan replied:
done

bstansberry reviewed (approved):
I had two final minor comments but this LGTM. Great work!

yersan commented:
Thank you both @tomjenkinson and @bstansberry for your time helping on this. I've pushed up the changes, I'll read it again and run it again to ensure code and blog are aligned. Then I'll squash the commits and close this PR and I'll file a PR for the Azure docs.
